src/js/preshuffle.s

```diff
@@ -1398,14 +1415,20 @@ UpdateEnemyHP:
     ; Check to see if the monster we attacked is on the blocklist
     ; by looking to see if it has a name. If the name table hi addr is $00
     ; then we don't want to display it (even if it does have HP)
-    ldx ObjectNameId,y
-    lda EnemyNameTableHi,x
-    beq +
-      sty CurrentEnemySlot
-      lda ShouldRedisplayUI
-      ora #DRAW_ENEMY_STATS
-      sta ShouldRedisplayUI
-+   rts
+    lda ObjectNameId,y
+    ; The blocklist should never be empty, but if it somehow does, then this code would break horribly
+.assert ENEMY_NAME_BLOCKLIST_LEN > 0
```

Comment:
Would it be reasonable to just check for a zero in the index table (hi byte)?  IIUC we're already running on page 3d, so the `beq +` you had before should basically already be doing the right thing.

Where does that break down?